### PR TITLE
Greenfield: Rename API key redirect params; switch to POST body

### PR DESCRIPTION
**Pull request by dennisreimann**
This way it is clearer what to expect and it also make the parameter easier to consume.

### CODE REVIEW

**Review by Kukks — CHANGES_REQUESTED** [review]
Since it is a post (which I completely forgot), we should actually send these props in the body

**Comment by dennisreimann** [comment]
Boy, that escalated quickly

File: BTCPayServer/Controllers/ManageController.APIKeys.cs [hunk 1]
```diff
@@ -157,16 +156,17 @@ public async Task<IActionResult> AddApiKey()
                     }
 
                     //we have a key that is sufficient, redirect to a page to confirm that it's ok to provide this key to the app.
-                    return View("Confirm",
-                        new ConfirmModel()
+                    return View("ConfirmAPIKey",
```
Comment (Kukks) [inline BTCPayServer/Controllers/ManageController.APIKeys.cs:159]: Why a new view?

Comment (dennisreimann) [reply BTCPayServer/Controllers/ManageController.APIKeys.cs:159]: 1. we need to pass through the params in the form body vs. the URL query
2. preparation for adding the „create a new separate key“ button in addition to the confirm („reuse existing key“) button

**Comment by NicolasDorier** [comment]
tests do not pass

**Comment by dennisreimann** [comment]
The integration test failures seeem unrelated (rate providers), but I'm still unhappy about how this turned out:
https://github.com/btcpayserver/btcpayserver/blob/a37b442c9607986ab397101526573abc9a4c52c8/BTCPayServer/Views/Shared/PostRedirect.cshtml

I couldn't find a way to apply attributes conditionally and Razor wouldn't let me use the form `action` in addition to the `asp-` attributes.  Couldn't just `if` the opening tag either, so it ended up like this.

File: BTCPayServer.Tests/ApiKeysTests.cs [hunk 2]
```diff
@@ -132,13 +132,13 @@ public async Task CanCreateApiKeys()
...
-                var apiKeyRepo = s.Server.PayTester.GetService<APIKeyRepository>();
-
-                await TestApiAgainstAccessToken(results.Single(pair => pair.Key == "key").Value, tester, user,
-                    (await apiKeyRepo.GetKey(results.Single(pair => pair.Key == "key").Value)).GetBlob().Permissions);
+                // var apiKeyRepo = s.Server.PayTester.GetService<APIKeyRepository>();
+                //
+                // IEnumerable<KeyValuePair<string, string>> results = url.Split("?").Last().Split("&")
+                //     .Select(s1 => new KeyValuePair<string, string>(s1.Split("=")[0], s1.Split("=")[1]));
+                // await TestApiAgainstAccessToken(results.Single(pair => pair.Key == "key").Value, tester, user,
+                //     (await apiKeyRepo.GetKey(results.Single(pair => pair.Key == "key").Value)).GetBlob().Permissions);
```
Comment (dennisreimann) [inline BTCPayServer.Tests/ApiKeysTests.cs:141]: Not sure how to test thosee now that they are part of the POST body and we land on an external page.

File: BTCPayServer.Tests/SeleniumTester.cs [hunk 3]
```diff
@@ -308,8 +308,9 @@ public string CreateInvoice(string storeName, decimal amount = 100, string curre
             currencyEl.Clear();
             currencyEl.SendKeys(currency);
             Driver.FindElement(By.Id("BuyerEmail")).SendKeys(refundEmail);
-            Driver.FindElement(By.Name("StoreId")).SendKeys(storeName + Keys.Enter);
-            Driver.FindElement(By.Id("Create")).ForceClick();
+            Driver.FindElement(By.Name("StoreId")).SendKeys(storeName);
+            Driver.FindElement(By.Id("Create")).Click();
+
```
Comment (dennisreimann) [inline BTCPayServer.Tests/SeleniumTester.cs:313]: This is related to the ChromeDriver upgrade.

File: BTCPayServer.Tests/ApiKeysTests.cs (hunk 2 above)
Comment (dennisreimann) [reply BTCPayServer.Tests/ApiKeysTests.cs]: Re-added with 3fe7515bcae457e6f9eec5c3dd6655bf46ccd13d.

**Comment by dennisreimann** [comment]
Now the PR can be reviewed. The main goal was to update two things regarding the callback the consumer app receives:
- send parameters as POST body instead of as part of the URL query string
- renaming the parameters to be more clear: `user` became `userId`, `key` became `apiKey`.

While working on this I also made some minor UI updates to the confirm view:

confirm

The separate confirm view got introduced to prepare the addition of a third button for the "create a new api key" button use case, which I'd like to tackle in a separate PR.

Also added a non-javascript version for the post redirect view:

nojs

File: BTCPayServer/Controllers/HomeController.cs [hunk 4]
```diff
@@ -199,6 +200,18 @@ public IActionResult RecoverySeedBackup(RecoverySeedBackupViewModel vm)
...
+        public ActionResult PostRedirectCallbackTestpage(IFormCollection data)
+        {
+            var list = data.Keys.Aggregate(new Dictionary<string, string>(), (res, key) =>
+            {
+                res.Add(key, data[key]);
+                return res;
+            });
+            return Json(list);
+        }
+
```
Comment (dennisreimann) [inline BTCPayServer/Controllers/HomeController.cs:214]: I saw that there is the `#if DEBUG` conditional. If this also works for the tests or there is something similar for those, wee could use this here too.

Comment (Kukks) [reply BTCPayServer/Controllers/HomeController.cs:214]: you can use it there for tests

File: BTCPayServer/BTCPayServer.csproj [hunk 5]
```diff
@@ -57,6 +57,7 @@
       <PrivateAssets>all</PrivateAssets>
       <IncludeAssets>runtime; build; native; contentfiles; analyzers</IncludeAssets>
     </PackageReference>
+    <PackageReference Include="Selenium.WebDriver.ChromeDriver" Version="85.0.4183.8700" />
```
Comment (Kukks) [inline BTCPayServer/BTCPayServer.csproj:60]: remove please

File: BTCPayServer/wwwroot/swagger/v1/swagger.template.authorization.json [hunk 6]
```diff
@@ -86,7 +86,7 @@
                         }
                     },
                     "307": {
-                        "description": "Redirects to the specified url in `redirect` with query string values for `key` (the api key created or matched), `permissions` (the permissions the user consented to), and `user` (the id of the user that consented) upon consent"
+                        "description": "Redirects to the specified url in `redirect` with query string values for `apiKey` (the api key created or matched), `permissions` (the permissions the user consented to), and `userId` (the id of the user that consented) upon consent"
```
Comment (Kukks) [inline BTCPayServer/wwwroot/swagger/v1/swagger.template.authorization.json:89]:
```suggestion
                        "description": "Makes browser do an HTTP POST request to the specified url in `redirect` with a JSON body consisting of `apiKey` (the api key created or matched), `permissions` (the permissions the user consented to), and `userId` (the id of the user that consented) upon consent"
```

**Review by Kukks — CHANGES_REQUESTED** [review]

File: BTCPayServer/Controllers/HomeController.cs (hunk 4 above)
Comment (dennisreimann) [reply BTCPayServer/Controllers/HomeController.cs]: Added, thanks!

**Review by Kukks — APPROVED** [review]

**Pull request merged**